### PR TITLE
KOGITO-2140 - Serverless Workflow - add support for sending event on …

### PR DESCRIPTION
**Pull request by tsurdilo**
…state transitions

The Serverless Workflow Specification added ability to produce event on state transition: https://github.com/cncf/wg-serverless/commit/d29fbd11c5376966552a9d02ea2039e8d03c0720

This PR adds the parsing and runtime support for this added feature in the spec.

### CODE REVIEW

**Comment by tsurdilo** [comment]
Jenkins seems to be having problems with:

[INFO] Running io.quarkus.it.kogito.drools.HotReloadTest
May 10, 2020 10:24:02 PM io.quarkus.deployment.dev.DevModeMain start
ERROR: Quarkus dev mode failed to start in curation phase
.....
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.kie.kogito:kogito-quarkus-deployment:jar:8.0.0-SNAPSHOT
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:414)
	... 58 more

File: jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java [hunk 1]
```diff
@@ -69,6 +69,11 @@
     public static final String HTP_GROUPID = "groupid";
     public static final String HT_ACTORID = "actorid";
     public static final String RF_GROUP = "ruleflowgroup";
+    public static final String METADATA_TRIGGERREF = "TriggerRef";
```
Comment (ruromero) [inline jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java:72]: Metadata constants are already defined in the Metadata class
https://github.com/kiegroup/kogito-runtimes/blob/master/jbpm/jbpm-flow/src/main/java/org/jbpm/ruleflow/core/Metadata.java#L22

File: jbpm/jbpm-serverless-workflow/src/test/java/org/jbpm/serverless/workflow/WorkflowFactoryTest.java [hunk 2]
```diff
@@ -347,4 +347,22 @@ public void testRuleSetNodeDefaultValues() {
...
+    public void testSendEventNode() {
+        TestNodeContainer nodeContainer = new TestNodeContainer();
+
+        EventDefinition eventDefinition = new EventDefinition().withName("testEvent")
+                .withSource("testSource").withType("testType");
+
+        ActionNode actionNode = testFactory.sendEventNode(1L, eventDefinition, nodeContainer);
+        assertThat(actionNode).isNotNull();
+        assertThat(actionNode.getName()).isEqualTo("testEvent");
+        assertThat(actionNode.getMetaData("TriggerType")).isEqualTo("ProduceMessage");
```
Comment (ruromero) [inline jbpm/jbpm-serverless-workflow/src/test/java/org/jbpm/serverless/workflow/WorkflowFactoryTest.java:361]: Use constants in Metadata

**Comment by cristianonicolai** [comment]
@tsurdilo same comments as @ruromero regarding constants otherwise looks good.

**Comment by tsurdilo** [comment]
@cristianonicolai @ruromero 
rebased to get the metadata class and updated. thanks for review.

**Comment by tsurdilo** [comment]
@cristianonicolai @tiagodolphine hi :) does this need more reviews or is it good to go ? Jenkins fails seem unrelated to the code changes.

**Comment by cristianonicolai** [comment]
@tsurdilo happy to merge once build is green, looking into that.

**Pull request merged**